app/jobs/delete_host.rb

```diff
@@ -23,10 +23,18 @@ def perform(message)
 
   def remove_related(host_id)
     num_removed = 0
+    to_rescore = Set.new # Keep a set of profiles that need to be rescored
     Sidekiq.logger.info("Deleting related records for host #{host_id}")
     MODELS.each do |model|
-      num_removed += model.where(host_id: host_id).delete_all
+      to_remove = model.where(host_id: host_id)
+      # Mark profile IDs as to be rescored if the model is TestResult
+      to_rescore.merge(to_remove.pluck(:profile_id)) if model == TestResult
```

Comment:
Yeah, that didn't feel too DRY to me, but I'll change :+1:

Reply:
So here's the thing, the order in deletion is important if we don't want to have DB integrity problems, so the test results should be deleted AFTER the rule results and the policy hosts links should be the last (I guess). It would not fire an error as the validations/callbacks aren't running, but it's not healthy as it's no longer wrapped in a transaction.

Reply:
Yes, that's a good point :laughing: I can see how this refactor might be difficult. I wouldn't block this PR on it - we can always go back and refactor this code; it's at least contained within this single file.

Reply:
Just warning you @akofink , the solution will be a bit esoteric :laughing: